Performance/Benchmarks/Benchmarks/GRPCSwiftBenchmark/Benchmarks.swift

```diff
@@ -14,24 +14,128 @@
  * limitations under the License.
  */
 import Benchmark
-import Foundation
+import GRPCCore
 
 let benchmarks = {
-    Benchmark.defaultConfiguration = .init(
-        metrics: [
-            .mallocCountTotal,
-            .syscalls,
-            .readSyscalls,
-            .writeSyscalls,
-            .memoryLeaked,
-            .retainCount,
-            .releaseCount,
-        ]
-    )
-    
-    // async code is currently still quite flaky in the number of retain/release it does so we don't measure them today
-    var configWithoutRetainRelease = Benchmark.defaultConfiguration
-    configWithoutRetainRelease.metrics.removeAll(where: { $0 == .retainCount || $0 == .releaseCount })
-    
-    // Add Benchmarks here
+  Benchmark.defaultConfiguration = .init(
+      metrics: [
+          .mallocCountTotal,
+          .syscalls,
+          .readSyscalls,
+          .writeSyscalls,
+          .memoryLeaked,
+          .retainCount,
+          .releaseCount,
+      ]
+  )
+  
+  // async code is currently still quite flaky in the number of retain/release it does so we don't measure them today
+  var configWithoutRetainRelease = Benchmark.defaultConfiguration
+  configWithoutRetainRelease.metrics.removeAll(where: { $0 == .retainCount || $0 == .releaseCount })
+  
+  Benchmark("Metadata - Add string") { benchmark in
+    for _ in benchmark.scaledIterations {
+      var metadata = Metadata()
+      for i in 0..<1000 {
+          metadata.addString("\(i)", forKey: "\(i)")
+      }
+    }
+  }
+  
+  Benchmark("Metadata - Remove string") { benchmark in
```

Comment:
`removeAllValues(forKey:)` doesn't care about the type being removed. We can remove the binary equivalent to this test and just call it "Remove all values for key" (or something along those lines)